File: backend/app/models/llm.py
from __future__ import annotations

import enum

from sqlalchemy import JSON, Boolean, Enum, Float, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base, TimestampMixin


class LlmProviderType(str, enum.Enum):
    openai = "openai"
    azure_openai = "azure_openai"
    deepseek = "deepseek"
    claude = "claude"
    ollama = "ollama"
    qwen = "qwen"
    custom = "custom"


class LlmProvider(Base, TimestampMixin):
    __tablename__ = "llm_providers"

    name: Mapped[str] = mapped_column(String(64), nullable=False, index=True)
    provider: Mapped[LlmProviderType] = mapped_column(
        Enum(LlmProviderType), nullable=False
    )
    model: Mapped[str] = mapped_column(String(128), nullable=False)
    base_url: Mapped[str | None] = mapped_column(String(255), nullable=True)
    api_key: Mapped[str | None] = mapped_column(Text, nullable=True)  # 加密后存储
    temperature: Mapped[float] = mapped_column(Float, default=0.2, nullable=False)
    max_tokens: Mapped[int | None] = mapped_column(Integer, nullable=True)
    extra_params: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    is_default: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    enabled: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    owner_id: Mapped[int] = mapped_column(
        ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True
    )

File: frontend/src/api/llm.ts
import http from './http'

export type LlmProviderType =
  | 'openai'
  | 'azure_openai'
  | 'deepseek'
  | 'claude'
  | 'ollama'
  | 'qwen'
  | 'custom'

export interface LlmProvider {
  id: number
  name: string
  provider: LlmProviderType
  model: string
  base_url?: string | null
  temperature: number
  max_tokens?: number | null
  is_default: boolean
  enabled: boolean
  has_api_key: boolean
  owner_id: number
}

export const llmApi = {
  list: () => http.get<unknown, LlmProvider[]>('/llm/providers'),
  create: (data: Partial<LlmProvider> & { api_key?: string }) =>
    http.post<unknown, LlmProvider>('/llm/providers', data),
  update: (id: number, data: Partial<LlmProvider> & { api_key?: string }) =>
    http.put<unknown, LlmProvider>(`/llm/providers/${id}`, data),
  remove: (id: number) => http.delete(`/llm/providers/${id}`),
  test: (id: number, prompt: string) =>
    http.post<unknown, { ok: boolean; output?: string; error?: string }>(
      `/llm/providers/${id}/test`,
      { prompt }
    )
}

export interface PromptTemplate {
  id: number
  name: string
  description?: string | null
  system_prompt: string
  keywords?: string[] | null
  filter_rules?: Record<string, unknown> | null
  output_schema?: Record<string, unknown> | null
  is_active: boolean
}

export const promptApi = {
  list: () => http.get<unknown, PromptTemplate[]>('/llm/prompts'),
  create: (data: Partial<PromptTemplate>) =>
    http.post<unknown, PromptTemplate>('/llm/prompts', data),
  update: (id: number, data: Partial<PromptTemplate>) =>
    http.put<unknown, PromptTemplate>(`/llm/prompts/${id}`, data),
  remove: (id: number) => http.delete(`/llm/prompts/${id}`)
}
